File: b2c-32-content-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function($http){
	//下拉品牌列表数据
	this.selectOptionList=function(){
		return $http.get('../brand/selectOptionList.do');
	}
	
	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../brand/findAll.do');		
	}

	//分页服务
	this.findPage=function(page,rows){
		console.log('brandService')
		return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);		
	}
	
	//保存服务 
	this.save=function(methodName,entity){
			return $http.post('../brand/'+ methodName +'.do',entity )
	}

	

	//查询单个服务 
	this.findOne=function(id){
		return $http.get('../brand/findOne.do?id='+id)				
	}
		 
	//批量删除服务 
	this.dele=function(selectIds){			
			//获取选中的复选框			
			return $http.get('../brand/delete.do?ids='+$scope.selectIds)
	}
		
	//条件查询服务 
	this.search=function(page,rows,searchEntity){
		console.log('brandService')
		return $http.post('../brand/search.do?page='+page+"&rows="+rows, searchEntity)			
	}
	



	
});

File: b2c-11-sellergoods-interface/src/main/java/com/jyr/sellergoods/service/BrandService.java
package com.jyr.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.jyr.pojo.TbBrand;

import entity.PageResult;

public interface BrandService {
	// 查询品牌
	public List<TbBrand> findAll();

	// 返回分页列表
	public PageResult findPage(int pageNum, int pageSize);

	// 增加品牌
	public void add(TbBrand brand);
	
	//查询一个品牌
	public TbBrand findOne(Long id);

	// 修改品牌
	public void update(TbBrand brand);
	
	//批量删除
	public void delete(Long [] ids);
	
	//条件查询
	public PageResult findPage(TbBrand brand, int pageNum,int pageSize);
	
	/**
	 * 品牌下拉框数据
	 */
	List<Map> selectOptionList();

}
